applicationset/generators/matrix.go

```diff
@@ -61,11 +61,11 @@ func (m *MatrixGenerator) GenerateParams(appSetGenerator *argoprojiov1alpha1.App
 
 			if appSet.Spec.GoTemplate {
 				tmp := map[string]interface{}{}
-				if err := mergo.Merge(&tmp, a); err != nil {
-					return nil, fmt.Errorf("failed to merge params from the first generator in the matrix generator with temp map: %w", err)
+				if err := mergo.Merge(&tmp, b, mergo.WithOverride); err != nil {
+					return nil, fmt.Errorf("failed to merge params from the second generator in the matrix generator with temp map: %w", err)
 				}
-				if err := mergo.Merge(&tmp, b); err != nil {
-					return nil, fmt.Errorf("failed to merge params from the first generator in the matrix generator with the second: %w", err)
+				if err := mergo.Merge(&tmp, a, mergo.WithOverride); err != nil {
+					return nil, fmt.Errorf("failed to merge params from the second generator in the matrix generator with the first: %w", err)
```

Comment:
I think it's more intuitive to have `b` take precedence over `a`. I believe that more closely matches the non-gotemplate behavior.

Reply:
Gotcha, okay, we can worry about precedence on this issue. https://github.com/argoproj/argo-cd/issues/14556 Let's stay as close as possible to current behavior.

Reply:
Let me add that I switched positions of `a` and `b` in this code to ensure _the same merge order_ as before. I tested in our local Argo and was surprised that I needed to do this. But it is actually the same with `merge` and `mergeOverwrite` functions in Helm: you need to change the parameter order to ensure the same behaviour (see https://masterminds.github.io/sprig/dicts.html#merge-mustmerge).